shell/browser/ui/inspectable_web_contents_view.cc

```diff
@@ -5,12 +5,250 @@
+  raw_ptr<views::View> view_;
+  raw_ptr<views::Widget> widget_;
+  ui::ImageModel icon_;
+};
+
+}  // namespace
+
 InspectableWebContentsView::InspectableWebContentsView(
     InspectableWebContents* inspectable_web_contents)
-    : inspectable_web_contents_(inspectable_web_contents) {}
+    : inspectable_web_contents_(inspectable_web_contents),
+      devtools_web_view_(new views::WebView(nullptr)),
+      title_(u"Developer Tools") {
+  if (!inspectable_web_contents_->is_guest() &&
+      inspectable_web_contents_->GetWebContents()->GetNativeView()) {
+    auto* contents_web_view = new views::WebView(nullptr);
+    contents_web_view->SetWebContents(
+        inspectable_web_contents_->GetWebContents());
+    contents_web_view_ = contents_web_view;
+  } else {
+    no_contents_view_ = new views::Label(u"No content under offscreen mode");
+  }
+
+  devtools_web_view_->SetVisible(false);
+  AddChildView(devtools_web_view_.get());
+  AddChildView(GetContentsView());
+}
+
+InspectableWebContentsView::~InspectableWebContentsView() {
+  if (devtools_window_)
+    inspectable_web_contents()->SaveDevToolsBounds(
+        devtools_window_->GetWindowBoundsInScreen());
+}
+
+void InspectableWebContentsView::SetCornerRadii(
+    const gfx::RoundedCornersF& corner_radii) {
+  // WebView won't exist for offscreen rendering.
+  if (contents_web_view_) {
+    contents_web_view_->holder()->SetCornerRadii(corner_radii);
+  }
+}
+
+void InspectableWebContentsView::ShowDevTools(bool activate) {
+  if (devtools_visible_)
+    return;
+
+  devtools_visible_ = true;
+  if (devtools_window_) {
+    devtools_window_web_view_->SetWebContents(
+        inspectable_web_contents_->GetDevToolsWebContents());
+    devtools_window_->SetBounds(inspectable_web_contents()->dev_tools_bounds());
+    if (activate) {
+      devtools_window_->Show();
+    } else {
+      devtools_window_->ShowInactive();
+    }
+
+    // Update draggable regions to account for the new dock position.
+    if (GetDelegate())
+      GetDelegate()->DevToolsResized();
+  } else {
+    devtools_web_view_->SetVisible(true);
+    devtools_web_view_->SetWebContents(
+        inspectable_web_contents_->GetDevToolsWebContents());
+    devtools_web_view_->RequestFocus();
+    DeprecatedLayoutImmediately();
+  }
+}
+
+void InspectableWebContentsView::CloseDevTools() {
+  if (!devtools_visible_)
+    return;
+
+  devtools_visible_ = false;
+  if (devtools_window_) {
+    auto save_bounds = devtools_window_->IsMinimized()
+                           ? devtools_window_->GetRestoredBounds()
+                           : devtools_window_->GetWindowBoundsInScreen();
+    inspectable_web_contents()->SaveDevToolsBounds(save_bounds);
+
+    devtools_window_.reset();
+    devtools_window_web_view_ = nullptr;
+    devtools_window_delegate_ = nullptr;
+  } else {
+    devtools_web_view_->SetVisible(false);
+    devtools_web_view_->SetWebContents(nullptr);
+    DeprecatedLayoutImmediately();
```

Comment:
[Chromium docs](https://chromium.googlesource.com/chromium/src/+/main/docs/ui/learn/bestpractices/layout.md?pli=1#don_t-invoke-deprecatedlayoutimmediately) seems to suggest using `InvalidateLayout` instead. You'll probably want to test this change manually to see if it has any effect.
```suggestion
    InvalidateLayout();
```

There's also two other instances of this method being called in this file.

Reply:
So this is original code from the removed `shell/browser/ui/views/inspectable_web_contents_view_views.cc` which was introduced in the: https://github.com/electron/electron/pull/41204/commits/cbdf43f9ec1993d91f6b87c069d67530a856b6a9

This is old cross platform code, so IMO better leave it as it is in this change which already introduces a lot of changes and potential regressions and create separate ticket for checking whether replacing `DeprecatedLayoutImmediately` to `InvalidateLayout` is safe.